examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,247 @@
+from transformers.modeling_bart import BartForConditionalGeneration
+from utlis import CnnDailyMailDataset, add_generic_args
+
+
+class BartSystem(pl.LightningModule):
+    def __init__(self, hparams):
+        super(BartSystem, self).__init__()
+        self.hparams = hparams
+        self.bart = BartForConditionalGeneration.from_pretrained("bart-large", output_past=True)
+        # self.rouge = Rouge()
+
+        self.tokenizer = BartTokenizer.from_pretrained("bart-large")
+        self.criterion = torch.nn.CrossEntropyLoss()
+
+    def forward(self, input_ids, attention_mask=None, decoder_input_ids=None, decoder_attention_mask=None):
+        return self.bart(
+            input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+        )
+
+    def training_step(self, batch, batch_idx):
+        outputs = self.forward(
+            batch["source_ids"], attention_mask=batch["source_mask"], decoder_input_ids=batch["target_ids"]
+        )
+
+        out = outputs[0]
+
+        x = F.log_softmax(out, dim=-1)
+        y = batch["target_ids_y"]
+        norm = (y != self.tokenizer.pad_token_id).data.sum()
+
+        targets = y.clone()
+        targets[y == self.tokenizer.pad_token_id] = -100
+        loss = self.criterion(x.contiguous().view(-1, x.size(-1)), targets.contiguous().view(-1)) / norm
+
+        tensorboard_logs = {"train_loss": loss}
+        return {"loss": loss, "log": tensorboard_logs}
+
+    def validation_step(self, batch, batch_idx):
+        outputs = self.forward(
+            batch["source_ids"], attention_mask=batch["source_mask"], decoder_input_ids=batch["target_ids"]
+        )
+        out = outputs[0]
+
+        x = F.log_softmax(out, dim=-1)
+
+        y = batch["target_ids_y"]
+        norm = (y != self.tokenizer.pad_token_id).data.sum()
+        targets = y.clone()
+        targets[y == self.tokenizer.pad_token_id] = -100
+
+        loss = self.criterion(x.contiguous().view(-1, x.size(-1)), targets.contiguous().view(-1)) / norm
+        return {"val_loss": loss}
+
+    def validation_end(self, outputs):
+        avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
+        tensorboard_logs = {"val_loss": avg_loss}
+        return {"avg_val_loss": avg_loss, "log": tensorboard_logs}
+
+    def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, second_order_closure=None):
+        # if self.trainer.use_tpu:
+        #     xm.optimizer_step(optimizer)
+        # else:
+        optimizer.step()
+        optimizer.zero_grad()
+        self.lr_scheduler.step()
+
+    def configure_optimizers(self):
+        t_total = len(self.train_dataloader())
+        no_decay = ["bias", "LayerNorm.weight"]
+        optimizer_grouped_parameters = [
+            {
+                "params": [p for n, p in self.named_parameters() if not any(nd in n for nd in no_decay)],
```

Comment:
(nit) why identical code twice? If not identical please comment :)

Reply:
What identical code?